docking/pose_dock.py

```diff
@@ -170,29 +165,34 @@ def pose_dock(
             raise (RuntimeError('invalid `transition_type` (must be one of ["insertion", "deletion"])'))
 
         calc_work = True
-        for step, lamb in enumerate(new_lambda_schedule):
-            ctxt.step(lamb)
-            if step % 100 == 0:
-                report.report_step(ctxt, step, lamb, box, bps, impls, guest_name, n_steps, 'pose_dock')
-                host_coords = ctxt.get_x_t()[: len(host_conf)] * 10
-                guest_coords = ctxt.get_x_t()[len(host_conf) :] * 10
-                report.write_frame(host_coords, host_mol, guest_coords, guest_mol, guest_name, outdir, step, 'pd')
-            if step in (0, int(n_steps/2), n_steps-1):
-                if report.too_much_force(ctxt, lamb, box, bps, impls):
-                    calc_work = False
-                    break
+        # (ytz): we gotta figure out how to batch this code, tbd: batch this
+
+        # collect a du_dl calculation every step
+        subsample_freq = 1
```

Comment:
@schmolly I wasn't sure how to work this in batches - so I just do the check at the last frame. Is this okay?

Reply:
Yep!